src/win32/fs.cpp

```diff
@@ -1777,6 +1765,7 @@ bool WinDirAccess::dnext(LocalPath& /*path*/, LocalPath& nameArg, bool /*follows
           || (ffd.cFileName[1] && ((ffd.cFileName[1] != '.') || ffd.cFileName[2]))))
         {
             nameArg.localpath.assign(ffd.cFileName, wcslen(ffd.cFileName));
+            nameArg.prependWithSeparator(globbase, gWfsa.localseparator);
```

Comment:
if (!globbase.empty()) ?

Reply:
Yes, I think you are right, thanks very much.